src/test/scala/org/constellation/consensus/RoundManagerTest.scala

```diff
@@ -0,0 +1,286 @@
+  }
+
+  test("it should not start a new block-creation round if another round is in progress") {
+    val roundInfo = mock[RoundInfo]
+    roundInfo.startedByThisNode shouldReturn true
+
+    roundManager.underlyingActor.rounds += RoundId("a") -> roundInfo
+
+    roundManager ! StartNewBlockCreationRound
+
+    roundManager.underlyingActor.rounds.size shouldBe 1
+  }
+
+  test("it should start a new block-creation round if another round is not in progress") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.rounds.size shouldBe 1
+    }
+  }
+
+  test("it should pass NotifyFacilitators to parent if new round has been started") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.passToParentActor(any[NotifyFacilitators]) was called
+    }
+  }
+
+  test("it should pass LightTransactionsProposal to round actor if new round has been started") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.passToRoundActor(any[LightTransactionsProposal]) was called
+    }
+  }
+
+  test("it should participate in a new round on ParticipateInBlockCreationRound") {
+    val roundData = new RoundData(
+      RoundId("round1"),
+      Set(peerData1, peerData2),
+      Set(),
+      FacilitatorId(facilitatorId1),
+      Seq(),
+      Seq(),
+      Seq()
+    )
+    val cmd = mock[ParticipateInBlockCreationRound]
+    cmd.roundData shouldReturn roundData
+
+    roundManager ! cmd
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.rounds.size shouldBe 1
+    }
+  }
+
+  test(
+    "it should pass StartTransactionProposal to round manager if has participated in a new round"
+  ) {
+    val roundData = new RoundData(
+      RoundId("round1"),
+      Set(peerData1, peerData2),
+      Set(),
+      FacilitatorId(facilitatorId1),
+      Seq(),
+      Seq(),
+      Seq()
+    )
+    val cmd = mock[ParticipateInBlockCreationRound]
+    cmd.roundData shouldReturn roundData
+
+    roundManager ! cmd
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.passToRoundActor(any[StartTransactionProposal]) was called
+    }
+  }
+
+  test("it should pass LightTransactionsProposal to round actor") {
+    val cmd = mock[LightTransactionsProposal]
+    cmd.roundId shouldReturn RoundId("round1")
+
+    roundManager ! cmd
+
+    roundManager.underlyingActor.passToRoundActor(cmd) was called
+  }
+
+  test("it should send ResolveMajorityCheckpointBlock to round actor when round timeout has passed") {
+    val timersRoundManagerProbe = TestProbe()
+    val timersRoundManager: TestActorRef[RoundManager] =
+      TestActorRef(Props(spy(new RoundManager(5 second), true)), timersRoundManagerProbe.ref)
+
+    timersRoundManager ! StartNewBlockCreationRound
+
+    Thread.sleep(10000)
```

Comment:
What do you think about making ` within(n seconds)` instead of thread sleep ?

Reply:
it doesn't work with `within(n seconds)` statement

Reply:
anyway, I've reduced it to 5 seconds